### ISSUE TITLE
Unable to configure deep dependencies

### ISSUE THREAD

**Issue by matthewp** [stealjs/steal#1111]
If you depend on a package `foo` that depends on a package `bar`, and you need to configure `bar` for some reason, you cannot do so.

The problem is that `bar` is a contextual module identifier for the `foo` package and there's no current way to declare the contextual relationship of module identifiers in configuration.

**Comment by frank-dspeed** [comment]
@matthewp please give a more complet example i don't understand the issue but i want't to think about a solution as i am good in such things.

**Comment by frank-dspeed** [comment]
@matthewp when i understood that right i can always create a third yoo module that requires configured bar and puts that in foo and exports it as yoo it represents foo and bar configured.

**Comment by frank-dspeed** [comment]
@matthewp if we talk about configuring it in steal options then i can simply add bar via package json and configure it?

**Comment by matthewp** [comment]
An example would be:

```
node_modules/foo <- depends on bar
node_modules/bar
```

If you depend on foo and foo depends on bar, but you don't depend on bar directly, you currently cannot configure bar.

**Comment by frank-dspeed** [comment]
@matthewp i can't use map nothing?

```
    "map": {
      "justifiedGallery": "justifiedGallery/src/js/justifiedGallery"
    },
    "meta": {
      "justifiedGallery/src/js/justifiedGallery": {
        "format": "global",
        "deps": [
          "jquery",
          "justifiedGallery/src/less/justifiedGallery.less"
        ]
      }
    },
```

**Comment by matthewp** [comment]
That works fine, but you couldn't add a map configuration for justifiedGallery's dependencies, if it had any.

**Comment by frank-dspeed** [comment]
@matthewp then i would say if a sub dependency needs configuration we can simply add it to package.json via npm install --save bar and configure it so that when foo requests it it gets the right configured bar ?

**Comment by matthewp** [comment]
Yeah, that's a workaround today. This issue is to figure out how not to need the workaround.

**Comment by frank-dspeed** [comment]
@matthewp thats easy its impossible by design :)

as steal gets a config you can forget that. we have no way to modify a require inside of a script.

**Comment by frank-dspeed** [comment]
@matthewp but i think this issue is also steal-tools related in case of packaging as soon as the slim loader is there we can maybe hook something into that
